### ISSUE TITLE
Relink `/usr/local/lib/libtensorflow_framework.so.2' with `/lib/x86_64-linux-gnu/libz.so.1' for IFUNC symbol `crc32_z'

### ISSUE THREAD

**Issue by suth1807** [tensorflow/tensorflow#41080]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version: libtensorflow nightly build
- Python version: 
- Installed using virtualenv? pip? conda?:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source): 9.3.0
- CUDA/cuDNN version: 10.1/7.6.5
- GPU model and memory: NVIDIA Quadro RTX5000

I have a trained model which I want to execute from my c++ program. My c++ program is a big piece of software that also uses google protocol buffers/ Intel MKL / Intel IPP / Boost. Linking goes through when I link to libtensorflow c API. However, when I run my exe I get a segmentation fault with the message given in subject line. If I write a separate program with libtensorflow without using any of the other dependencies mentioned above, things work fine.

1) Downloaded nightly build of libtensorflow (other older versions also lead to the same crash)
2) compiled and linked my main program. Execution results in crash
3) Execution works fine if I run with a sample code without linking to other libraries mentioned above

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by Saduf2019** [comment]
@suth1807 
Please refer to [this link](https://stackoverflow.com/questions/48306849/lib-x86-64-linux-gnu-libz-so-1-version-zlib-1-2-9-not-found) and let us know if it helps.

**Comment by suth1807** [comment]
Thank you @Sadug2019, I could find the root cause of the problem. It is
actually the use of  Protocolbuffer library both in my application and in
libtensorflow.so. I am guessing there are static initializations in
libprotobuf  that happen twice. This leads to this mysterious message and a
crash (zlib is already on my system). I also tried to build protocol
buffers without zlib but that didn't solve the crash (The message went away
though). The only workaround I have is to load libtensorflow in my
application with dlopen with RTLD_DEEPBIND flag. This solves the crash and
works fine but I still think it is a workaround. My problem is the same as
mentioned in
https://github.com/tensorflow/tensorflow/issues/24976 (I found the
RTLD_DEEPBIND trick from there). Is there a possibility of resolving this
correctly?
